.github/workflows/e2e-main.yaml

```diff
@@ -0,0 +1,143 @@
+#
+# Copyright (C) 2025 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: e2e-tests-main
+
+on:
+  push:
+    branches: [main]
+
+  workflow_dispatch:
+    inputs:
+      organization:
+        default: 'minc-org'
+        description: 'Organization of the Minc Extension repository'
+        type: string
+        required: true
+      repositoryName:
+        default: 'minc-extension'
+        description: 'Minc Extension repository name'
+        type: string
+        required: true
+      branch:
+        default: 'main'
+        description: 'Minc Extension repo branch'
+        type: string
+        required: true
+
+jobs:
+  e2e-tests:
+    name: Run E2E tests
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v5
+        if: github.event_name == 'workflow_dispatch'
+        with:
+          repository: ${{ github.event.inputs.organization }}/${{ github.event.inputs.repositoryName }}
+          ref: ${{ github.event.inputs.branch }}
+          path: ${{ github.event.inputs.repositoryName }}
+
+      - uses: actions/checkout@v5
+        if: github.event_name == 'push'
+        with:
+          path: minc-extension
+
+      # Checkout podman desktop
+      - uses: actions/checkout@v5
+        with:
+          repository: podman-desktop/podman-desktop
+          ref: main
+          path: podman-desktop
+
+      - uses: actions/setup-node@v5
+        with:
+          node-version: 22
+
+      - uses: pnpm/action-setup@v4
+        name: Install pnpm
+        with:
+          run_install: false
+          package_json_file: ./podman-desktop/package.json
+
+      - name: Update podman
```

Comment:
please reuse custom gh action for updating podman. https://github.com/redhat-actions/podman-install